app/src/main/java/com/sun/chat_04/ui/signup/SignUpActivity.kt

```diff
@@ -13,32 +13,101 @@ import android.view.View
 import com.jpardogo.android.googleprogressbar.library.ChromeFloatingCirclesDrawable
+import com.sun.chat_04.R
 import com.sun.chat_04.R.color
 import com.sun.chat_04.R.id
 import com.sun.chat_04.R.layout
 import com.sun.chat_04.data.model.User
-import kotlinx.android.synthetic.main.sign_up_screen.button_signUp
-import kotlinx.android.synthetic.main.sign_up_screen.edit_confirm_password
-import kotlinx.android.synthetic.main.sign_up_screen.edit_email
-import kotlinx.android.synthetic.main.sign_up_screen.edit_fullname
-import kotlinx.android.synthetic.main.sign_up_screen.edit_password
+import com.sun.chat_04.data.remote.UserRemoteDataSource
+import com.sun.chat_04.data.repositories.UserRepository
+import kotlinx.android.synthetic.main.sign_up_screen.buttonSignUp
+import kotlinx.android.synthetic.main.sign_up_screen.editConfirmPassword
+import kotlinx.android.synthetic.main.sign_up_screen.editEmail
+import kotlinx.android.synthetic.main.sign_up_screen.editFullname
+import kotlinx.android.synthetic.main.sign_up_screen.editPassword
 import kotlinx.android.synthetic.main.sign_up_screen.progress
-import kotlinx.android.synthetic.main.sign_up_screen.radio_groups
-import kotlinx.android.synthetic.main.sign_up_screen.text_birthday
+import kotlinx.android.synthetic.main.sign_up_screen.radioGroups
+import kotlinx.android.synthetic.main.sign_up_screen.textBirthday
 import kotlinx.android.synthetic.main.sign_up_screen.toolbar
 import java.util.Calendar
 
-class SignUpActivity : AppCompatActivity(), OnClickListener {
-    private var user: User? = null
+class SignUpActivity : AppCompatActivity(), OnClickListener, SignUpContract.View {
+
+    private var presenter: SignUpContract.Presenter? = null
+    private var user: User = User()
+    private var email = ""
+    private var password = ""
+    private var confirmPassword = ""
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(layout.sign_up_screen)
+        val firebaseAuth = FirebaseAuth.getInstance()
+        val firebaseDatabase = FirebaseDatabase.getInstance()
+        presenter = SignUpPresenter(this, UserRepository(UserRemoteDataSource(firebaseAuth, firebaseDatabase)))
         initComponents()
         requirePermissions()
     }
 
+    override fun onClick(v: View?) {
+        when (v?.id) {
+            id.textBirthday -> showDatePicker()
+            id.buttonSignUp -> handleSignUp()
+        }
+    }
+
+    override fun onSignUpFailure() {
+        showNotify(resources.getString(R.string.sign_up_success))
+    }
+
+    override fun onSignUpSuccessfuly() {
+        showNotify(resources.getString(R.string.sign_up_existed))
+    }
+
+    override fun onEmptyUserName() {
+        showNotify(resources.getString(R.string.sign_up_emptyUserName))
+        progress.visibility = View.GONE
+    }
+
+    override fun onEmptyBirthday() {
+        showNotify(resources.getString(R.string.sign_up_emptyBirthday))
```

Comment:
```suggestion
        showNotify(resources.getString(R.string.sign_up_error_empty_birth_day))
```
check các string khác nữa